src/main/Config.cpp

```diff
@@ -321,14 +322,37 @@ Config::load(std::string const& filename)
                 }
                 ALLOW_LOCALHOST_FOR_TESTING = item.second->as<bool>()->value();
             }
-            else if (item.first == "MAINTENANCE_ON_STARTUP")
+            else if (item.first == "AUTOMATIC_MAINTENANCE_PERIOD")
             {
-                if (!item.second->as<bool>())
+                if (!item.second->as<int64_t>())
+                {
+                    throw std::invalid_argument(
+                        "invalid AUTOMATIC_MAINTENANCE_PERIOD");
+                }
+                int64_t parsedAutomaticMaintenancePeriod =
+                    item.second->as<int64_t>()->value();
+                if (parsedAutomaticMaintenancePeriod < 0 ||
+                    parsedAutomaticMaintenancePeriod > UINT32_MAX)
```

Comment:
style: use braces around the `throw` (especially as the if is already spanning multiple lines)